x-pack/filebeat/input/awss3/input.go

```diff
@@ -129,10 +130,12 @@ func (in *s3Input) Run(inputContext v2.Context, pipeline beat.Pipeline) error {
 		}
 		defer receiver.metrics.Close()
 
-		// Poll sqs waiting metric periodically in the background.
+		// Poll metrics periodically in the background
 		go pollSqsWaitingMetric(ctx, receiver)
+		metricReporterChan := make(chan int64)
+		go pollSqsUtilizationMetric(ctx, receiver, metricReporterChan)
```

Comment:
Can you encapsulate the details around the SQS utilization into `inputMetrics`. Ideally the signature of `sqsReader.Receive` would not need to change because `sqsReader` already holds a reference to the `*inputMetrics` instance.